src/event.rs

```diff
@@ -661,6 +661,16 @@ impl<'a, ES: EventSet> Sender<'a, ES> {
         unsafe { self.world.queue_global(ptr, GlobalEventIdx(event_idx)) };
     }
 
+    /// Grab the state of the sender; the [`EventSet`] indices.
+    pub fn state(&self) -> &ES::Indices {
+        self.state
+    }
+
+    /// Grab the world cell.
+    pub fn world(&self) -> &UnsafeWorldCell<'a> {
+        &self.world
+    }
```

Comment:
This is currently unsound because `UnsafeWorldCell::world` is safe to call. Mark `UnsafeWorldCell::world` as unsafe and we should be good. `UnsafeWorldCell` should also be made into a `HandlerParam` at this point.